### PR TITLE
#486 make MP Config compilable with Java11

### PR DESCRIPTION
**Pull request by Emily-Jiang**
Signed-off-by: Emily Jiang <emijiang6@googlemail.com>

### CODE REVIEW

**Comment by Emily-Jiang** [comment]
This closes #486

**Comment by Emily-Jiang** [comment]
@dmlloyd please take a look

File: tck/src/main/java/org/eclipse/microprofile/config/tck/ConfigProviderTest.java [hunk 1]
```diff
@@ -155,13 +155,15 @@ public void testInjectedConfigSerializable() {
         ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
         try (ObjectOutputStream out = new ObjectOutputStream(byteArrayOutputStream)) {
             out.writeObject(config);
-        } catch (IOException ex) {
```
Comment (kenfinnigan) [inline tck/src/main/java/org/eclipse/microprofile/config/tck/ConfigProviderTest.java]: Are the formatting changes in this and `ConverterTest` required for JDK 11?

Comment (dmlloyd) [reply tck/src/main/java/org/eclipse/microprofile/config/tck/ConfigProviderTest.java]: I was thinking the same, but just assumed it was due to checkstyle.  But if not, then the formatting changes should be left out probably...

Comment (kenfinnigan) [reply tck/src/main/java/org/eclipse/microprofile/config/tck/ConfigProviderTest.java]: +1

Comment (Emily-Jiang) [reply tck/src/main/java/org/eclipse/microprofile/config/tck/ConfigProviderTest.java]: It was to sastisfy the checkstyle.

**Comment by kenfinnigan** [comment]
Why the checkstyle rule change?

I don’t think we want open braces on the next line

**Comment by pzygielo** [comment]
Which modification is the key to make MPC Java11 compilable? Is CS upgrade and related changes necessary to accomplish this goal? On my machine upgrading just the `maven-javadoc-plugin` does the job.

**Comment by Emily-Jiang** [comment]
@kenfinnigan the new version of checkstyle requires the closing bracket is on the new line.
@pzygielo I updated the javadoc version together with others to move up the version.

**Comment by pzygielo** [comment]
@Emily-Jiang 
>  I updated the javadoc version together with others to move up the version.

I know, but it is out of the scope of commit message. It seems like other changes are required, which they are not.

**Comment by kenfinnigan** [comment]
@Emily-Jiang then the rule needs to be adjusted. We shouldn't be changing the code style due to upgrading the version of checkstyle

**Comment by Emily-Jiang** [comment]
> Why the checkstyle rule change?
I don’t think we want open braces on the next line

The checkstyle has a backward changes from 2.17 to 3.1. I had a look at the [option](https://checkstyle.sourceforge.io/property_types.html#rcurly) for right curly braces.
alone, alone_or_singleline, same
In the version 2.17, we used 'alone`, which does not enforce alone.
I tried to change to same. If I do, I need to change a few more files.

Basically, we cannot maintain the inconsistent state anymore. @kenfinnigan it is not open braces, but close braces. I think it is ok.

**Comment by kenfinnigan** [comment]
Ah yeah, close braces. My preference is not to have a closing brace on a different line than any `catch`.

I'm leaning towards an earlier comment that we need to revert checkstyle updates and just make the changes needed for JDK 11, and revisit checkstyle issues later.

Ideally the whole of MP uses the same checkstyle rules.

**Comment by Emily-Jiang** [comment]
@kenfinnigan I've reverted the checkstyle upgrade. I've raised [this](https://github.com/eclipse/microprofile/issues/159) issue and discuss further on the MP project level.

**Pull request merged**